damnit/backend/listener.py

```diff
@@ -73,20 +78,30 @@ def __init__(self, context_dir=Path('.')):
         # Fail fast if read-only - https://stackoverflow.com/a/44707371/434217
         self.db.conn.execute("pragma user_version=0;")
         self.proposal = self.db.metameta['proposal']
-        log.info(f"Will watch for events from proposal {self.proposal}")
 
-        if gethostname().startswith('exflonc'):
-            # running on the online cluster
-            kafka_conf = KAFKA_CONF['onc']
-        else:
-            kafka_conf = KAFKA_CONF['maxwell']
+        log.info(f"Will watch for events from proposal {self.proposal}")
 
         consumer_id = CONSUMER_ID.format(self.db.metameta['db_id'])
-        self.kafka_cns = KafkaConsumer(*kafka_conf['topics'],
-                                       bootstrap_servers=kafka_conf['brokers'],
+        self.kafka_cns = KafkaConsumer(*KAFKA_TOPICS,
+                                       bootstrap_servers=KAFKA_BROKERS,
                                        group_id=consumer_id)
-        self.events = kafka_conf['events']
 
+        # check backend host and connection to online cluster
+        self.online_data_host = None
+        self.run_online = self.db.metameta.get('run_online', False) is not False
+        if self.run_online and gethostname() not in BACKEND_HOSTS_TO_ONLINE:
+            log.warning(f"Disabled online processing, the backend must run on one of: {BACKEND_HOSTS_TO_ONLINE}")
+            self.run_online = False
+        if self.run_online:
+            topic = Path(find_proposal(f'p{self.proposal:06}')).parts[-3]
+            if (remote_host := ONLINE_HOSTS.get(topic)) is None:
+                log.warn(f"Can't run online processing for topic '{topic}'")
```

Comment:
```suggestion
                log.warning(f"Can't run online processing for topic '{topic}'")
```

The log 'warn' alias is deprecated, although it will probably never actually be removed.